### DISCUSSION TITLE
Weekly update 2020-11-12

### DISCUSSION THREAD

**Discussion by frol** [near/nearcore#3597] [Node Public Interfaces]
## Current focus

* Indexer: @khorolets: Refactored the Indexer for Explorer DB schema to be easier to navigate through. Indexers for Explorer for testnet and mainnet are redeployed. TODO: Idempotency of accounts and access keys storage needs to be addressed to enable high-availability setup (running multiple nodes at once).

## Plans

* Rosetta RPC: @frol: fix bugs, implement the view-calls as of the latest Rosetta spec, contruction API inlcuded in the rosetta.cfg

## Discussions

* @chefsale: IO is too high when the node is under high load of RPC calls (Mario has added a detailed logging per method name, so we are about to learn about the specifics) -- needs further investigation (/cc @bowenwang1996)

**Comment by frol** [comment]
One more thing: I had a discussion with ETC Labs on [OpenRPC](https://open-rpc.org) support in nearcore as a way to define our JSON-RPC API as OpenRPC spec, and we will consider writing that spec in 2021 Q1.

**Reply by ilyar** [reply]
I wonder how things are with it?

**Reply by frol** [reply]
@ilyar Hey, no progress has been made on that side since OpenRPC tooling did not progress any further, and even a year from now it remains to be a very niche project with only JS/TS client generation which basically defeats the purpose since we neither can generate the specs from our Rust implementation, nor generate clients from the spec (even if we go with writing it by hands) for the languages that we don't support these days.

**Reply by ilyar** [reply]
thanks for the answer
![image](https://user-images.githubusercontent.com/761285/134820695-88b9c03a-ca9a-4e8e-927f-6aaeec3a1cac.png)
I noticed this tool on the homepage has a claim that it works. But when I started looking, I found only this topic. I think it will be useful to update the section "Who is building on NEAR?".

**Reply by frol** [reply]
Well, OpenRPC folks do some steps on their side (they promised Rust integrations and more client languages support), so eventually, that will get shaped out just like TheGraph and Covalent (we also work with them closely). Probably, we can put a label on those projects that are not released yet, but we don't want to remove them completely.

**Comment by bowenwang1996** [comment]
> @chefsale: IO is too high when the node is under high load of RPC calls (Mario has added a detailed logging per method name, so we are about to learn about the specifics) -- needs further investigation (/cc @bowenwang1996)

@frol could we create an issue for this?

**Reply by frol** [reply]
@bowenwang1996 I feel that is quite a history now. We have the monitoring in place now and if we ever observe it again, we will investigate further.
